### DISCUSSION TITLE
Current visitors

### DISCUSSION THREAD

**Discussion by samuel-git** [umami-software/umami#1269] [General]
Thank you for this wonderful tool and all the hard work going into it.

I noticed a discrepancy between the current visitor stats. E.g.: 
* 20 current visitors reported on realtime page
* 54 current visitors reported in dashboard
* 80 visitors connected via active WebSockets to our backend

The actual number is likely around 2-5 times higher than the reported number in the dashboard as not every page on our website uses WebSockets.

**Comment by mikecao** [comment]
The realtime and dashboard should be the same. Maybe a bug so we can look into it.

Websocket connections may not be the same because the user can still be connected to the socket but fall off the current users window (5 minutes).

**Reply by peterhanania** [reply]
I've also realized the current visitor stat is inaccurate.

- If I login from a desktop, it showed me 42 current visitors and stuck at that number
- If I login from mobile, it showed me 22 current visitors and stuck there too
- If I go to realtime on desktop I see different stats from the current visitors
- If I go to realtime on mobile I see a lot more numbers than the desktop version

I don't know if its a caching issue or something. Was this fixed?

**Comment by samuel-git** [comment]
This explains the websocket discrepancy. I am using v1.30.0 of Umami with PostgreSQL.
The realtime current visitors are usually 2-3 times lower than the ones reported on the dashboard.
Definitely not a real issue as long as the daily visitors in the dashboard are correct.
Lot's of people stay longer than 5 minutes as our average time on site indicates, but again, no real issue for us.
Umami works way better than other solutions.

**Comment by gandhis1** [comment]
I would think there'd be a way to do some kind of `setTimeout` stuff and have it re-push the presence of a client on an interval. As long as someone as a web page open, there should be some statistic that shows them as an active visitor. That could be a separate stat from the 5-minute rolling window or not.
